PY2 catharsis

alimanfoo opened the pull request:
This PR removes support for Python 2.7.

TODO:
* [x] Check docs for any mention of Python 2.
* [x] Release notes.

alimanfoo commented:
Hi @hardingnj, decided to go ahead with dropping PY2 support, give ourselves one less headache going forwards. Some documentation work still TODO but I all code changes should be there. Think I removed all the PY2 code but would be grateful if you could give this a once over.

hardingnj commented on tox.ini:
```diff
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
```
I'm not familiar with `tox`, but line 32 suggests tox uses 2.7. Unsure if this is an issue?

hardingnj commented:
Happy to make those changes by the way and PR (just unsure if they are oversights or have a purpose!)

Docs look clean of python 2 references by the way. (apart from release notes, which is obvs fine)

alimanfoo commented:
Thanks @hardingnj for the review on this. I've fixed the items you mentioned and added a release note. Going to merge if CI passes. 

cc @leehart for interest, I started this PR a while back but didn't get round to finishing it at the time, just wrapping it up now.
